code/game/objects/structures/kitchen_spike.dm

```diff
@@ -110,12 +110,12 @@
 		if(!M.buckled)
 			return
 		var/mob/living/L = buckled_mob
-		var/matrix/m120 = matrix(L.transform)
+		var/matrix/m120 = matrix()
 		m120.Turn(360)

```

Comment:
I think this is the issue actually.
turn 360 is going back to where you started, isn't it?